### PR TITLE
Adding hot reload to server

### PR DESCRIPTION
**Pull request by corey-nm**
Adds cli flags to deepsparse.server, which calls the file_watcher introduced in #660.

Test plan: unit tests and running locally.

### CODE REVIEW

File: src/deepsparse/server/cli.py [hunk 1]
```diff
@@ -90,9 +90,20 @@ def main():
     default="info",
     help="Sets the logging level.",
 )
-def config(config_path: str, host: str, port: int, log_level: str):
+@click.option(
+    "-r",
+    "--hot-reload-config",
+    is_flag=True,
+    default=False,
+    help="Hot reload the config whenever the file is updated.",
```
Comment (bfineran) [inline src/deepsparse/server/cli.py:98]: prefer to be a little more descriptive here, ie saying deployed routes will be updated based on latest config

File: src/deepsparse/server/cli.py [hunk 2]
```diff
@@ -171,6 +182,13 @@ def config(config_path: str, host: str, port: int, log_level: str):
         "a default integration such as Prometheus."
     ),
 )
+@click.option(
+    "-r",
+    "--hot-reload-config",
+    is_flag=True,
+    default=False,
+    help="Hot reload the config whenever the file is updated.",
```
Comment (bfineran) [inline src/deepsparse/server/cli.py:190]: ""

File: src/deepsparse/server/server.py [hunk 3]
```diff
@@ -64,6 +70,10 @@ def start_server(
     server_config = ServerConfig(**obj)
     _LOGGER.info(f"Using config: {repr(server_config)}")
 
+    if hot_reload_config:
+        _LOGGER.info(f"Watching {config_path} for changes.")
+        _ = start_config_watcher(config_path, f"http://{host}:{port}/endpoints", 0.5)
```
Comment (bfineran) [inline src/deepsparse/server/server.py:75]: we can probably afford to check less frequently than 0.5s no?

File: tests/server/test_server.py [hunk 4]
```diff
@@ -0,0 +1,104 @@
+# Copyright (c) 2021 - present / Neuralmagic, Inc. All Rights Reserved.
```
Comment (bfineran) [inline tests/server/test_server.py:1]: why are we creating a new `test_server.py` just for hot_config_reload functionality?

**Review by bfineran — COMMENTED** [review]
LGTM, minor comments

File: tests/server/test_server.py (hunk 4 above)
Comment (corey-nm) [reply tests/server/test_server.py:1]: this one is more of an integration test that actually spins up a server process via `start_server`. probably could even move this to test the cli tbh. felt different 🤷‍♂️

File: src/deepsparse/server/server.py (hunk 3 above)
Comment (corey-nm) [reply src/deepsparse/server/server.py:75]: It's pretty cheap to check even every half second. Prefer a more responsive watcher

File: src/deepsparse/server/cli.py (hunk 1 above)
Comment (corey-nm) [reply src/deepsparse/server/cli.py:98]: Updated

File: tests/server/test_server.py (hunk 4 above)
Comment (corey-nm) [reply tests/server/test_server.py]: Move to test_hot_reloading.py

**Review by KSGulin — DISMISSED** [review]

File: src/deepsparse/server/cli.py [hunk 5]
```diff
@@ -90,9 +90,23 @@ def main():
     default="info",
     help="Sets the logging level.",
 )
-def config(config_path: str, host: str, port: int, log_level: str):
+@click.option(
+    "-r",
```
Comment (bfineran) [inline src/deepsparse/server/cli.py:94]: nit: no other options have a shortcut

**Review by bfineran — DISMISSED** [review]

File: src/deepsparse/server/cli.py (hunk 5 above)
Comment (corey-nm) [reply src/deepsparse/server/cli.py]: Removed

**Review by bfineran — APPROVED** [review]

**Pull request merged**